blob/service.go

```diff
@@ -211,83 +211,38 @@ func (s *Service) getByCommitment(
 		}
 		return nil, nil, err
 	}
+
 	for _, row := range namespacedShares {
-		if len(row.Shares) == 0 {
-			break
+		appShares, err := toAppShares(row.Shares...)
+		if err != nil {
+			return nil, nil, err
 		}
-
-		rawShares = append(rawShares, row.Shares...)
+		rawShares = append(rawShares, appShares...)
 		proofs = append(proofs, row.Proof)
 
-		// reconstruct the `blobShare` from the first rawShare in range
-		// in order to get blob's length(first share will contain this info)
-		if blobShare == nil {
-			for i, shr := range rawShares {
-				bShare, err := shares.NewShare(shr)
-				if err != nil {
-					return nil, nil, err
-				}
-
-				// ensure that the first share is not a NamespacePaddingShare
-				// these shares are used to satisfy the non-interactive default rules
-				// and are not the part of the blob, so should be removed.
-				isPadding, err := bShare.IsPadding()
-				if err != nil {
-					return nil, nil, err
-				}
-				if isPadding {
-					continue
-				}
-
-				blobShare = bShare
-				// save the length.
-				length, err := blobShare.SequenceLen()
-				if err != nil {
-					return nil, nil, err
-				}
-				amount = shares.SparseSharesNeeded(length)
-				rawShares = rawShares[i:]
-				break
-			}
-		}
-
-		// move to the next row if the blob is incomplete.
-		if amount > len(rawShares) {
-			continue
-		}
-
-		blob, same, err := constructAndVerifyBlob(rawShares[:amount], commitment)
+		var blobs []*Blob
+		blobs, rawShares, err = createBlobs(rawShares)
 		if err != nil {
 			return nil, nil, err
 		}
-		if same {
-			return blob, &proofs, nil
+		for _, b := range blobs {
+			if b.Commitment.Equal(commitment) {
+				return b, &proofs, nil
+			}
+			if expanded {
+				// only the first blob in the range can be expanded
```

Comment:
what do you mean by this? It's not possible that another blob in the set of blobs by namespace can overflow the row?

Reply:
No, because we are going row by row appending row.Shares to the shares from the previous rows(if they have existed), and they could be existed **only** if the blob spans multiple rows

Reply:
Okay but what do you mean by "only the first blob in the range" ?